indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -231,6 +231,15 @@ def _check_access(self):
                  'reject': self.booking.can_reject,
                  'cancel': self.booking.can_cancel}
 
+        # The `can_*` methods contain both access and state checks, but it's nicer to show
+        # something more verbose in case e.g. someone accepted a conflicting booking which
+        # auto-rejected another one, and they then try to reject that from another tab.
+        match self.action:
+            case 'approve' if not self.booking.is_pending:
+                raise ExpectedError(_('This booking has already been approved or rejected.'))
```

Comment:
Is it possible to split this into two messages? One for approved bookings and one for rejected bookings? I think it would look a bit nicer to users when they this message.

Reply:
fair enough :+1: